lxl-web/src/lib/components/Search.svelte

```diff
@@ -54,10 +66,23 @@
 		}
 		return undefined;
 	}
+
+	function handleTransform(data) {
+		// hijacking this function to set get autosuggest labels
+		// to not hardcode lxl-needs into useSearchRequest
+		// could also be new separate callback
+		suggestMapping = data?.mapping;
+		return data;
+	}
+
+	function getLabels(key: string, value?: string) {
+		let pageMapping = $page.data.searchResult?.mapping;
+		return getLabelFromMappings(key, value, pageMapping, suggestMapping);
+	}
```

Comment:
This could be rewritten into a derived state:
```js
let derivedLxlQualifierPlugin = $derived.by(() => {
	function getLabels(key: string, value?: string) {
		let pageMapping = $page.data.searchResult?.mapping;
		return getLabelFromMappings(key, value, pageMapping, suggestMapping);
	}
	return lxlQualifierPlugin((getLabels));
});
```
Which can then be used further down in the file (where the extensions are set on the `<SuperSearch>` component):
```extensions={[derivedLxlQualifierPlugin]}```


It results in two more lines of code but can maybe clarify the scoping of `getLabels` and `lxlQualifierPlugin`.

What do you think?

Reply:
Nice, looks cleaner!
Is the double parenthethis in the return statement intentional?

Reply:
Applied this ☝️

Reply:
> Is the double parentheses in the return statement intentional?

Oups, not at all! 🙃